metrics/web/lib/common/domain/entities/instant_config.dart

```diff
@@ -0,0 +1,31 @@
+import 'package:meta/meta.dart';
+
+/// An entity representing an instant config.
+@immutable
+class InstantConfig {
+  /// Indicates whether to show the login form.
+  final bool isLoginFormEnabled;
+
+  /// Indicates whether the FPS monitor feature is enabled.
+  final bool isFpsMonitorEnabled;
+
+  /// Indicates whether the renderer display feature is enabled.
+  final bool isRendererDisplayEnabled;
+
+  /// Creates a new instance of the [InstantConfig].
+  ///
+  /// Throws an [ArgumentError] if either of the given [isLoginFormEnabled],
+  /// [isFpsMonitorEnabled] or [isRendererDisplayEnabled] is `null`.
+  InstantConfig({
+    this.isLoginFormEnabled,
+    this.isFpsMonitorEnabled,
+    this.isRendererDisplayEnabled,
+  }) {
+    ArgumentError.checkNotNull(isLoginFormEnabled, 'isLoginFormEnabled');
+    ArgumentError.checkNotNull(isFpsMonitorEnabled, 'isFpsMonitorEnabled');
+    ArgumentError.checkNotNull(
```

Comment:
Not sure we have to check these values here as we have the logic with default in the use case, which means that it is possible for the repository to return some of these values as `null`. Instead, I would suggest making appropriate parameters required for the `InstantConfigParam`.